pup-0003.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Cherry-Picking Commits
+author: David Davis
+created: 01-Apr-2017
+status:
+---
+
+## Summary
+
+Currently in Pulp, we merge features to master. However, hotfixes and bug fixes
+are merged to release branches which then are merged to newer release branches
+and then to master (a process called "merging forward"). More info about
+merging forward can be found in our documentation:
+
+http://docs.pulpproject.org/en/nightly/dev-guide/contributing/merging.html#merging-your-changes
+
+Also, a good illustration of this process is outlined in the following webpage:
+
+http://nvie.com/posts/a-successful-git-branching-model/
+
+This PUP is to use cherry-picking instead of merging forward. Using this model,
+all PRs would be made against master and then changes would be cherry-picked to
+release branches as needed.
```

Comment:
There is some info about hotfix and release branches [here](https://docs.pulpproject.org/dev-guide/contributing/building.html#when-to-build-from-a-dev-branch-a-tag-or-hotfix-branch).
_My understanding_ is that right now if hotfix is needed, we create a separate branch from the <current>-release branch, cherry-pick what is needed and when hotfix is released its branch will be merged forward to <current>-release, then to <current>-dev, then to master. The only special case is when <current>-dev contains only the fixes needed for hotfix, hotfix release can be build from this branch then.